### ISSUE TITLE
[no-inferrable-types] Change defaults? + incorrect documentation

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#822]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-inferrable-types": "error",
  }
}
```

```TS
class Foo {
  prop: number = 5;
}

function fn(a: number = 5, b: boolean = true) {}
```

**Expected Result**

Errors for inferrable types on properties and parameters.

**Actual Result**

No errors.

**Additional Info**

This is because `ignoreParameters` and `ignoreProperties` are `true` by default, which is inconsistent with TSLint* and seems like an odd choice.

`*` I know that you're not aiming for 1:1 with TSLint, but I see no reason to change the defaults in this case.

Also, the documentation says that the above code is `invalid` for the default options, but it's not.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
hmmm - it looks like this is actually a bug introduced 8 months ago by me misinterpreting the base config

https://github.com/bradzacher/eslint-plugin-typescript/pull/261/files#diff-b3d257023fda2c7c79d1f23aab9ddf7c
